packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx

```diff
@@ -191,15 +192,17 @@ export const UploadControls = (): JSX.Element => {
 
   React.useEffect(() => {
     if (!initialRun.current) {
-      if (selected.type === 'UPLOAD_FILES') {
+      if (selected.files) {
+        handleFileSelect(selected.files);
+      } else if (selected.type === 'UPLOAD_FILES') {
         handleFileInput.current?.click();
       } else if (selected.type === 'UPLOAD_FOLDER') {
         handleFolderInput.current?.click();
       }
```

Comment:
This changes the behavior so dropping files on the Location Detail view will navigate to the Upload Action view and add the files to the queue without the file picker opening

Reply:
What happens here if i select the `Upload Files` action from the location detail action list?

Reply:
If you select `Upload Files` from the action list, it goes to the Upload Action view and opens the file picker

Reply:
Ah, missed the `else if` in the diff